ui/src/components/NamespaceSelectorList/index.tsx

```diff
@@ -0,0 +1,86 @@
+import { Check, Circle, Loader2, Square } from "lucide-react";
+import {
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+  CommandStaticItem,
+} from "~/design/Command";
+
+import { twMergeClsx } from "~/util/helpers";
+import { useListNamespaces } from "~/api/namespaces/query/get";
+import { useNamespace } from "~/util/store/namespace";
+import { useTranslation } from "react-i18next";
+
+type NamespaceSelectorListProps = {
```

Comment:
Regarding

> the array selectedValues is needed when isMultiSelect is true and not used when not. In the TS declaration, both are optional because I couldn't figure out how to declare that dependency without running into trouble later. Suggestions welcome on how to improve this.

This is what I came up with, I hope this is what you had in mind

```typescript
type MultiSelectorProps = { isMultiSelect: true; selectedValues?: string[] };
type SingleSelectorProps = { isMultiSelect?: false; selectedValues?: never };

type NamespaceSelectorListProps = {
  onSelectNamespace: (value: string) => void;
} & (MultiSelectorProps | SingleSelectorProps);
``` 

These are my testcases

```typescript
const Test = () => (
  <div>
    {/* multiselect no selected values defined */}
    <NamespaceSelectorList onSelectNamespace={() => null} isMultiSelect />

    {/* multiselect selected values defined */}
    <NamespaceSelectorList
      onSelectNamespace={() => null}
      isMultiSelect
      selectedValues={[]}
    />

    {/* single select */}
    <NamespaceSelectorList onSelectNamespace={() => null} />

    {/* single select with ts error when accidentally passing selectedValues */}
    <NamespaceSelectorList onSelectNamespace={() => null} selectedValues={[]} />
  </div>
);
```

Reply:
Awesome, I used this (but with selectedValues required when isMultiSelect is true).